### ISSUE TITLE
Can't output summaries using tensorflow.contrib.slim

### ISSUE THREAD

**Issue by mxmxlwlw** [tensorflow/tensorflow#6519]
When I run my code, it gives the following warning:
`WARNING:tensorflow:From C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\training\supervisor.py:344 in __init__.: SummaryWriter.__init__ (from tensorflow.python.training.summary_io) is deprecated and will be removed after 2016-11-30.
Instructions for updating:
Please switch to tf.summary.FileWriter. The interface and behavior is the same; this is just a rename.`

And It can't output the summaries. Although I think it is training my net. Here's some of my code:

`summaries = set(tf.get_collection(tf.GraphKeys.SUMMARIES))
for end_point in end:
    x=end[end_point]
    summaries.add(tf.summary.histogram('activations/' + end_point, x))
    summaries.add(tf.summary.scalar('sparsity/' + end_point,  tf.nn.zero_fraction(x)))
for loss in tf.get_collection(tf.GraphKeys.LOSSES):
    summaries.add(tf.summary.scalar('losses/%s' % loss.op.name, loss))
for variable in slim.get_model_variables():
     summaries.add(tf.summary.histogram(variable.op.name, variable))
summaries.add(tf.summary.scalar('learning_rate', learning_rate))
summaries.add(tf.summary.scalar('total_loss', total_loss))
#summaries |= set(tf.get_collection(tf.GraphKeys.SUMMARIES))
summary_op = tf.summary.merge(list(summaries))
slim.learning.train(train_op, 
    model_store_dir, 
    init_fn = init_fn,
    summary_op=summary_op,
    number_of_steps=100000,
    save_summaries_secs=300, 
    save_interval_secs=600)`

**Comment by inflation** [comment]
@mxmxlwlw As it said, SummaryWriter has renamed to `tf.summary.FileWriter`

**Comment by mxmxlwlw** [comment]
@inflation I noticed that slim.learning.train has a parameter summary_writer. So I passed the tf.summary.FileWriter to it, and the warning didn't show again. However, the log still did't come up. I think the error may be in other place of my code.
Here's my code:

import pickle
import re
import random
import numpy as np
import tensorflow as tf
from vgg_preprocessing import preprocess_image
slim = tf.contrib.slim
_R_MEAN = 123.68
_G_MEAN = 116.78
_B_MEAN = 103.94
def vgg16Net(inputs,
           num_classes=1000,
           is_training=True,
           dropout_keep_prob=0.5,
           spatial_squeeze=True,
           scope='vgg_16'):
    with tf.variable_scope(scope, 'vgg_16', [inputs]) as sc:
        end_points_collection = sc.name + '_end_points'
        # Collect outputs for conv2d, fully_connected and max_pool2d
        with slim.arg_scope([slim.conv2d, slim.fully_connected, slim.max_pool2d],
                            outputs_collections=end_points_collection):
          net = slim.repeat(inputs, 2, slim.conv2d, 64, [3, 3], scope='conv1')
          net = slim.max_pool2d(net, [2, 2], scope='pool1')
          net = slim.repeat(net, 2, slim.conv2d, 128, [3, 3], scope='conv2')
          net = slim.max_pool2d(net, [2, 2], scope='pool2')
          net = slim.repeat(net, 3, slim.conv2d, 256, [3, 3], scope='conv3')
          net = slim.max_pool2d(net, [2, 2], scope='pool3')
          net = slim.repeat(net, 3, slim.conv2d, 512, [3, 3], scope='conv4')
          net = slim.max_pool2d(net, [2, 2], scope='pool4')
          net = slim.repeat(net, 3, slim.conv2d, 512, [3, 3], scope='conv5')
          net = slim.max_pool2d(net, [2, 2], scope='pool5')
          # Use conv2d instead of fully_connected layers.
          net = slim.conv2d(net, 4096, [7, 10], padding='VALID', scope='fc6')
          net = slim.dropout(net, dropout_keep_prob, is_training=is_training,
                             scope='dropout6')
          net = slim.conv2d(net, 4096, [1, 1], scope='fc7')
          net = slim.dropout(net, dropout_keep_prob, is_training=is_training,
                             scope='dropout7')
          net = slim.conv2d(net, num_classes, [1, 1],
                            activation_fn=None,
                            normalizer_fn=None,
                            scope='fc8')
          # Convert end_points_collection into a end_point dict.
          end_points = slim.utils.convert_collection_to_dict(end_points_collection)
          if spatial_squeeze:
            net = tf.squeeze(net, [1, 2], name='fc8/squeezed')
            end_points[sc.name + '/fc8'] = net
          return net, end_points

def getsamples():
    rootpath = 'C:\\Users\\mx\\Desktop\\nextLevel\\UCF-101'
    with open('res.pickle', 'rb') as f:
        pathdict = pickle.load(f)
    with open('trainlist01.txt', 'r') as f:
        lines=f.readlines()
    samples_all=[]
    labels_all=[]
    for line in lines:
        [videogroup, video, label]=re.split('/| |\n', line)[0:3]
        samples = [rootpath + '\\' + videogroup + '\\' + video[:-4] + '\\' + i for i in pathdict[videogroup][video]]
        samples_all.extend(samples)
        labels_all.extend([label]*len(samples))
    return samples_all, labels_all

samples_all, labels_all = getsamples()

labels_one_hot_all = np.zeros((len(labels_all), 101))
index_offset = np.arange(len(labels_all))*101
ind = index_offset + np.array(labels_all, np.int32) - 1
labels_one_hot_all.flat[ind]=1
#samples_all = tf.constant(samples_all)
#labels_all = tf.constant(labels_all)
[sample, label] = tf.train.slice_input_producer([samples_all, labels_one_hot_all])
imagecontent = tf.read_file(sample)
image = tf.image.decode_jpeg(imagecontent, channels=3)
image = tf.cast(image, dtype = tf.float32)
channels = tf.split(2, 3, image)
channels[0] -= _R_MEAN
channels[1] -= _G_MEAN
channels[2] -= _B_MEAN
image=tf.concat(2, channels)
image=tf.reshape(image, [240, 320, 3])
images, labels = tf.train.batch([image, label], 16, 4, 64)
net, end = vgg16Net(images, num_classes = 101, is_training=True)

slim.losses.softmax_cross_entropy(net, labels)
total_loss = slim.losses.get_total_loss()
learning_rate = 0.001
optimizer=slim.train.GradientDescentOptimizer(learning_rate)
train_var = slim.get_variables_to_restore(exclude = ['vgg_16/conv1', 'vgg_16/conv2', 'vgg_16/conv2', 'vgg_16/conv3', 'vgg_16/conv4', 'vgg_16/conv5']);
train_op=slim.learning.create_train_op(total_loss, optimizer, variables_to_train = train_var)
model_store_dir = './log/'
init_var = slim.get_variables_to_restore(exclude = ['vgg_16/fc6', 'vgg_16/fc7', 'vgg_16/fc8'])
init_fn = slim.assign_from_checkpoint_fn('./vgg_16.ckpt', init_var)

summaries = set(tf.get_collection(tf.GraphKeys.SUMMARIES))
for end_point in end:
    x=end[end_point]
    summaries.add(tf.summary.histogram('activations/' + end_point, x))
    summaries.add(tf.summary.scalar('sparsity/' + end_point, tf.nn.zero_fraction(x)))

for loss in tf.get_collection(tf.GraphKeys.LOSSES):
    summaries.add(tf.summary.scalar('losses/%s' % loss.op.name, loss))

for variable in slim.get_model_variables():
    summaries.add(tf.summary.histogram(variable.op.name, variable))

summaries.add(tf.summary.scalar('learning_rate', learning_rate))

summaries.add(tf.summary.scalar('total_loss', total_loss))

#summaries |= set(tf.get_collection(tf.GraphKeys.SUMMARIES))

summary_op = tf.summary.merge(list(summaries))

summary_writer = tf.summary.FileWriter(model_store_dir)
slim.learning.train(train_op, 
    model_store_dir, 
    init_fn = init_fn,
    summary_op=summary_op,
    summary_writer=summary_writer,
    number_of_steps=100000,
    save_summaries_secs=300, 
    save_interval_secs=600)

**Comment by mxmxlwlw** [comment]
@inflation Here's the code.

[doNet.zip](https://github.com/tensorflow/tensorflow/files/674192/doNet.zip)

**Comment by inflation** [comment]
You can try to add `tf.summary.FileWriter.add_summary` before your training

**Comment by mxmxlwlw** [comment]
@inflation It suggest that I can't pass tensor to the add_summary function...

**Comment by inflation** [comment]
Slim uses `supervisor` for training. Do you have the model checkpoint files?

**Comment by mxmxlwlw** [comment]
@inflation Yes, I downloaded it from slim page in tensorflow/models(repository)

**Comment by inflation** [comment]
No, I mean when you trying to train your model, does it produce anything in the output directory?

**Comment by mxmxlwlw** [comment]
@inflation No. So you mean it didn't train?

**Comment by inflation** [comment]
Yes, checkpoint files should be in the same folder. And they are automatically saved in some period when using supervisor if the path is valid.

**Comment by mxmxlwlw** [comment]
I noticed that it's not the problem of summary. The training process can't even do one process step. Here's my new python code. Can you find what's the problem in that file? I really don't know how to fix it. 
[doNet3.zip](https://github.com/tensorflow/tensorflow/files/679508/doNet3.zip)

**Comment by mxmxlwlw** [comment]
Problem solved. I loss the code tf.train.start_queue_runners(sess=sess). It needs to run queue manually.
